### ISSUE TITLE
DASH-HLS Interop - EXT-X-DATERANGE and EMSG

### ISSUE THREAD

**Issue by johnsim** [cta-wave/content-specification-task-force#34]
From review of Annex D. CMAF Encoding for DASH-HLS Interoperability

Cooper Pope, 3/20/2019: HLS has the EXT-X-DATERANGE tag for carrying timed metadata in media playlists -- I’m sure you know the DASH equivalent is EMSG. If you’re using CMAF with both manifest protocols, synchronizing messages between DATERANGE and EMSG would be important. Some non-Apple players already support EMSG with HLS-CMAF since their media engines have a DASH pedigree, but I’ll defer to Apple for comment on how the details in this doc should be communicated.

**Comment by ZmGorynych** [comment]
My $.02
In-manifest carriage: HLS `EXT-X-DATERANGE` is equivalent to DASH `Event` element
Inband carriage: DASH `emsg` functionality is implemented in HLS using ID3v2

**Comment by mcpv** [comment]
I was attempting to highlight the point that emsg boxes are not supported natively as inband Timed Metadata in CMAF/fmp4 content on iOS/tvOS right now. That is, they will not be passed through the Timed Metatdata API like ID3v2 would from a TS segment. Therefore, if you need access to emsg in HLS-CMAF (on Apple devices at least), I was suggesting they could be carried as DATERANGE in the playlist. Apologies for not being more specific.

**Comment by johndeu** [comment]
@johnsim  is this still open? I thought we published the  AOM spec for compatibility of emsg carriage for HLS already. Can we point to that?

**Comment by krasimirkolarov** [comment]
It is at 

https://github.com/AOMediaCodec/id3-emsg/

Krasimir

**Comment by nicoweilelemental** [comment]
The AOM spec solves only some of the problems, which is using emsg as the shared envelope. 

The other (unsolved) problems are related to the content of the message_data field of the emsg box:
- There is no support for ID3 in the DASH spec
- DASH section 5.10.4 defines a precise list of values and scheme URNs for specific use cases, for which there is no equivalent in ID3

So my question here are: 
- Which kind of messages do we exactly want to synchronize between HLS and DASH, for which use case?
- If in-manifest carriage is the only appropriate approach, how would we represent the same message in both HLS `EXT-X-DATERANGE` and DASH `Event` syntax?

**Comment by johndeu** [comment]
Is it required in the DASH spec to provide the exact list of supported URI's that would identify the payload of the message_data payload?  That would not make much sense. The goal I thought was to make the emsg box very flexible. You can define any "schema" URN/URL that you would want - and it is up to you to decide what the payload is.  

In this case, the AOM spec defines a URN for ID3, and how to interpret the payload. 
I done see a need for the DASH specification to need to reference that. 
Was the goal of 5.10.4 to provide an exhaustive list of all URN's that could be defined for 'emsg'?  That would seem like a major mistake to me if it was.

**Comment by jpiesing** [comment]
I'm confused about this discussion. 5.10.4 defines the events that are consumed by (targeted at) the DASH player. That's it. Anyone is free to define other event schemes and the DASH player should pass these through to the calling app.

**Comment by johndeu** [comment]
Exactly @jpiesing  That section is not supposed to cover the events that are "custom" defined.  Users can define any scheme with any payload. So ID3 does not need to be explicitly declared in the DASH spec, as the MPEG group did not own the ID3 schema anyways and could not declare the URI at the time of writing.  The AOM spec added the scheme URN for ID3, so it is now available to be used by anyone that wants to.

**Comment by johnsim** [comment]
I agree with @johndeu  and @jpiesing, but it does seem reasonable for there be a spec governing the use of ID3 in a mixed DASH-HLS environment - even if the spec just points to the AOM published ID3 spec for CMAF. This might be an issue to be covered in the WAVE DASH-HLS Interop spec.
